### PR TITLE
State management, file system cleanup

### PR DESCRIPTION
**Pull request by AnonymusRaccoon**
This PR introduces a way to handle the game state and scene switching in a safe context. After every update, a given method is run with two arguments: the engine (as a mutable value) and a mutable arbitrary state. This run method should be used soon for UI.

This PR also removes components and systems that were in the ECS to put them in the Bomberman scope (no pre-made component or system should exist in the ECS library). I also moved tests back to the main scope because I did not have the faith to create tests components/systems, I only moved them.

### CODE REVIEW

File: sources/Component/Movable/MovableComponent.cpp [hunk 1]
```diff
@@ -4,13 +4,13 @@
 
 #include "MovableComponent.hpp"
 
-namespace WAL
+namespace Bomberman
```
Comment (Octopus773) [inline sources/Component/Movable/MovableComponent.cpp:7]: Namespace name is BBM

File: sources/main.cpp [hunk 2]
```diff
@@ -17,19 +20,19 @@
 #include "Vector/Vector3.hpp"
 #include "Window.hpp"
 
-int main()
+int demo()
```
Comment (Octopus773) [inline sources/main.cpp:23]: :(

File: CMakeLists.txt [hunk 3]
```diff
@@ -9,8 +9,40 @@ include_directories(bomberman lib/wal/sources)
 add_subdirectory(${PROJECT_SOURCE_DIR}/lib/wal)
 add_subdirectory(${PROJECT_SOURCE_DIR}/lib/Ray)
 
+set(SOURCES
```
Comment (Octopus773) [inline CMakeLists.txt:12]: why aren't you using the normal technic
(add_executable wth all files)

Comment (AnonymusRaccoon) [reply CMakeLists.txt:12]: Because I do not want to copy paste all files.

File: sources/main.cpp (hunk 2 above)
Comment (AnonymusRaccoon) [reply sources/main.cpp:23]: What's wrong? Are you sad because you didn't eat a beignet?

Comment (Octopus773) [reply sources/main.cpp:23]: i'm not obelix Maadame !

Comment (Octopus773) [reply sources/main.cpp:23]: you will find your response [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)

**Review by GitBluub — APPROVED** [review]

**Review by Arthi-chaud — APPROVED** [review]

**Review by Octopus773 — APPROVED** [review]

**Pull request merged**